### ISSUE TITLE
Grapplers pulling out of a 5 point harness

### ISSUE THREAD

**Issue by Xpyder** [CleverRaven/Cataclysm-DDA#22764]
Game version: 0.C-25577

Operating system: Windows 7

Tiles or curses: Tiles

Mods active:
Dark Days ahead
filthy clothing
disable npc needs
simplified nutrition
makeshift items
more survival tools
alternative map key
folding parts pack
vehicle additions pack
tanks and other verhicles

### Expected behavior
Grappler zombie grabbing me at 40mph in 5-point harness should either have it's arms dislocated (take damage), or be dragged along with me.  Either way I should stay in the seat.

### Actual behavior
Magically pulls me out of the vehicle while taking no damage

### Steps to reproduce the behavior
On Powerbike (design like motorcycle, but with car parts, including full seat, 5-point harness, and v4 engine)
At 40mph
Pass grappler zombie
Die

**Comment by Xpyder** [comment]
TL:DR - I need help figuring how to get a list/collection of vehicle_part at a particular grid position (from a tripoint location)


this just happened to me 3 times in an hour so I put some work into trying to figure out how to resolve it.

While ideally it would drag the zombie along with you, I'm not sure how to do this so I'm currently trying to figure out to have wearing a seatbelt block it (with a possible overload for pull strength and different resistances offered by seatbelt vs 5-point)

I've been looking into modifying the monster abilities (ids and files below).  It seems like monsters, characters, players don't have a "Seatbelt" flag, we have to derive it by checking the vehicle parts on their tile.

using RANGED_PULL as an example I can figure out how to get the pull target's position
`tripoint position = target->pos()`

and I can figure out if a part has the seatbelt flag
```
vehicle_part part == something
bool part.has_flag(VPFLAG_SEATBELT)
```

But I can't figure out how to get a list of parts on a tripoint grid position.
I can get a list of Vehicles on a grid position (`VehicleList vehicles = get_vehicles(position,position)`) but as near as I can tell that would check ALL the parts in those vehicles, not just the tile the player is on.

Help figuring out the right functions (or how to derive the information manually) would be appreciated

monster abilities I found on grapplers:
BIO_OP_TAKEDOWN
\src\monattack.cpp
\src\monattack.h


RANGED_PULL
\doc\JSON_FLAGS.md
\src\monattack.cpp
\src\monattack.h

GRAB_DRAG
\src\monattack.cpp
\src\monattack.h
